snapcraft/_options.py

```diff
@@ -124,7 +124,68 @@ def _get_platform_architecture():
     return architecture
 
 
-class ProjectOptions:
+class ProjectInfo:
+    """Information gained from the snap's snapcraft.yaml file."""
+
+    def __init__(self, *,
+                 name: str, version: str,
+                 summary: str, description: str, confinement: str) -> None:
+        self.__name = name
+        self.__version = version
+        self.__summary = summary
+        self.__description = description
+        self.__confinement = confinement
+
+    @property
+    def name(self):
+        """Returns the name of the snap.
+        """
+        return self.__name
+
+    @property
+    def version(self):
+        """Returns the version of the snap.
+        """
+        return self.__version
+
+    @property
+    def summary(self):
+        """Returns the summary of the snap.
+        """
+        return self.__summary
+
+    @property
+    def description(self):
+        """Returns the description of the snap.
+        """
+        return self.__description
+
+    @property
+    def confinement(self):
+        """Returns the confinement type of the snap.
+        """
+        return self.__confinement
+
+
+class Project:
+    """All details around building a project concerning the build environment
+    and the snap being built."""
+
+    def __init__(self) -> None:
+        self.__info = None
+
+    @property
+    def info(self):
+        """Returns the project info.
+        """
+        return self.__info
+
+    @info.setter
```

Comment:
What is the goal, here? What do we gain beyond simply setting `self.info = None` within `__init__`? I'd rather see good-old functions, personally. Gives us an API that scales a bit better.

Reply:
As said in the PR description, info needs to be set after creating the instance. It's a setter because otherwise the property would be read-only.
But I'm happy to make it a method for clarity.